www/officers/coi.cgi

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env ruby
+PAGETITLE = "Conflict of Interest Affirmations"
+$LOAD_PATH.unshift '/srv/whimsy/lib'
+require 'wunderbar'
+require 'wunderbar/bootstrap'
+require 'whimsy/asf'
+require 'mail'
+require 'date'
+require 'tmpdir'
+
+coi_url = ASF::SVN.svnurl('conflict-of-interest')
+YEAR = DateTime.now.strftime "%Y"
+COI_CURRENT_URL = File.join(coi_url, YEAR)
+COI_CURRENT_TEMPLATE_URL = File.join(COI_CURRENT_URL, 'template.txt')
+
+user = ASF::Person.find($USER)
+USERID = user.id
+USERNAME = user.cn
+committees = ASF::Committee.officers + ASF::Committee.nonpmcs
+chairs = committees.map do |committee|
+ committee.chairs.map {|chair| chair[:id]}
+end
+IDS = (chairs.flatten + ASF::Service['board'].members.map(&:id)).uniq
+
+# Get the list of files in this year's directory
+signerfileslist, err = ASF::SVN.svn('list', COI_CURRENT_URL, {user: $USER, password: $PASSWORD})
+signerfiles = signerfileslist.split('\n')
+
+# Create the hash of {signer: signerurl} and remember user's affirmation file
+SIGNERS = Hash.new
+user_affirmation_file = nil
+signerfiles.each do |signerfile|
+  stem = signerfile[0..signerfile.index(".")-1]
+  user_affirmation_file = signerfile if stem == USERID
+  SIGNERS[stem] = signerfile unless stem == 'template'
```

Comment:
Does it matter if there is more than one file for a given stem?
Can it occur?

Reply:
Should not occur as long as the user is using this tool. If the user has already signed, the tool will not ask for another affirmation. So the only way this can happen is if the user uses svn directly which there is no reason to. But in any event, having any file with the user's id will prevent the tool from generating another one.